### ISSUE TITLE
Transform for the optional chaining affects code coverage

### ISSUE THREAD

**Issue by webschik** [alangpierce/sucrase#622]
Hello,

I found a problem with `sucrase` that doesn't allow my team to use it with `jest` :(
The code we get from an optional chanining transform cannot be fully covered by test cases and it leads to the drop of the code coverage results.

E.g. we have a source file
```ts

// test.ts

export default function test(data?: {value: number}) {
    return data?.value;
}
```

`sucrase` compiles it to
```js
// test.js

"use strict";
Object.defineProperty(exports, "__esModule", {value: true});
function _optionalChain(ops) {
    let lastAccessLHS = undefined;
    let value = ops[0];
    let i = 1;

    while (i < ops.length) {
        const op = ops[i];
        const fn = ops[i + 1];

        i += 2;
        if ((op === 'optionalAccess' || op === 'optionalCall') && value == null) {
            return undefined;
        }

        if (op === 'access' || op === 'optionalAccess') {
            lastAccessLHS = value; value = fn(value);
        } else if (op === 'call' || op === 'optionalCall') {
            value = fn((...args) => value.call(lastAccessLHS, ...args));
            lastAccessLHS = undefined;
        }
    }

    return value;
}

function test(data) {
    return _optionalChain([data, 'optionalAccess', _ => _.value]);
}

exports.default = test;
```

and the test

```js
// test.spec.js

import test from './test.ts';

describe('test()', () => {
    it('should return undefined', () => {
        expect(test()).toBeUndefined();
    });

    it('should return value', () => {
        expect(test({value: 42})).toBe(42);
    });
});

```

After running `jest` we get a code coverage like this

<img width="688" alt="sucrase_file_coverage" src="https://user-images.githubusercontent.com/1665314/121493845-7ea82600-c9d8-11eb-80dc-0fc31abf2a3d.png">

If it's not possible to fix on a compiler level, maybe it's possible to consider an opt-in functionality for some transforms? Maybe like a compilation target with an ES version?
Because we run the tests on Node v14.8 and optional chaining is supported there.

P.S. I'm not sure if I should have updated issue #602 or created a new one :)

**Comment by alangpierce** [comment]
Interesting, thanks for reporting. #602 has definitely been on my mind, and this is yet another argument for it. I left a comment there describing an approach that I'd accept as a PR.

**Comment by webschik** [comment]
@alangpierce I've opened PR #623 ;)

**Comment by webschik** [comment]
fixed in #623
